File: php/monsters/Hero.php
<?php

namespace monsters;

use monsters\options\Fightable;

use Random;

class Hero implements Fightable {
    private $score;
    private $random;

    public function __construct() {
        $this->random = new Random();
        $this->score = 0;
    }

    public function getScore() {
        return $this->score;
    }

    public function setScore($score) {
        $this->score = $score;
    }

    public function attack() {
        return $this->random->nextInt(100);
    }
}

?>

File: java/monsters/Hero.java
package monsters;

import monsters.options.Fightable;

import java.util.Random;

public class Hero implements Fightable {
    private int score;
    private final Random random;

    public Hero() {
        this.random = new Random();
        this.score = 0;
    }

    public int getScore() {
        return score;
    }

    public void setScore(int score) {
        this.score = score;
    }

    public int attack() {
        return random.nextInt(100);
    }
}
